src/uri/uri.cc

```diff
@@ -249,6 +249,34 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto parsed_path = URI{std::string{path}};
+  const std::string path_to_set = parsed_path.path().value();
+  const bool has_leading_slash = path_to_set.front() == '/';
+
+  this->path_ = has_leading_slash ? path_to_set.substr(1) : path_to_set;
+  return *this;
+}
+
+auto URI::path(std::string &&path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto parsed_path = URI{std::move(path)};
+  const std::string path_to_set = parsed_path.path().value();
+  const bool has_leading_slash = path_to_set.front() == '/';
+
+  this->path_ = has_leading_slash ? path_to_set.substr(1) : path_to_set;
```

Comment:
This is also still performing a copy. At the end of the day, I think this overload is much closer to the copy overload that it should be. Maybe there are ways to make it a proper move?

Reply:
For example: you are already parsing the path as a URI into `parsed_path`. At that point, it is already parsed by your parsing rules, so in theory you could just do this?

```
this->path_ = std::move(parsed_path.path());
```

?

if its `std::nullopt`, I think it would also handle the `if` above too

Reply:
Oh yeah I should continue to move the std::string as much as I can! Otherwise I'll have a move semantics that contains copy that is totally non sense!

Reply:
By the way this feedback works for both `path` methods